drivers/gpio/gpio_eos_s3.c

```diff
@@ -0,0 +1,432 @@
+/*
+ * Copyright (c) 2020, Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT quicklogic_eos_s3_gpio
+
+#include <errno.h>
+#include <drivers/gpio.h>
+#include <soc.h>
+#include <eoss3_hal_gpio.h>
+#include <eoss3_hal_pads.h>
+#include <eoss3_hal_pad_config.h>
+
+#include "gpio_utils.h"
+
+#define MAX_GPIOS_PER_PORT	8
```

Comment:
If you define this a `8U` you might get away from having to cast values to `(int)` below.